### ISSUE TITLE
Consider a YouTube search feed

### ISSUE THREAD

**Issue by xbc5** [xbc5/sumo#9]
Not all topics/people have a dedicated or official channel. Create a feature that subscribes to search results.

Some caveats:
- the results change often
- duplication of effort

For both caveats we should use a caching mechanism, whether it's the database, or the file system.  Regardless, this will result in a two-stage process, which may contradict the generic feed interface (#5). The feed interface will expect a one-shot, fetch and format process; but the two-stage process will require 1) perform the search, 2) for each result, check the cache, if it misses, then fetch. Additionally, this will make another request to pull the item using the scraper approach (#6).

In conclusion, #5 will need to abstract the search, caching mechanism, and the scraping of each item. Only then that we can create/emit the finished feed type.

In short: scrape YouTube search for URLs; filter only new items, then scrape each one.
